controller_manager:
  ros__parameters:
    update_rate: 100  # Hz
    
    # Controller names to be loaded
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
      
    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

joint_trajectory_controller:
  ros__parameters:
    joints:
      - shoulder_pitch
      - shoulder_roll
      - upper_arm_rotation
      - elbow_pitch
      - forearm_rotation
      - wrist_pitch
      - wrist_yaw
    
    command_interfaces:
      - position
    
    state_interfaces:
      - position
      - velocity
    
    state_publish_rate: 50.0
    action_monitor_rate: 20.0
    
    allow_partial_joints_goal: false
    constraints:
      goal_time: 0.5
      stopped_velocity_tolerance: 0.01
      shoulder_pitch:     {trajectory: 0.05, goal: 0.03}
      shoulder_roll:      {trajectory: 0.05, goal: 0.03}
      upper_arm_rotation: {trajectory: 0.05, goal: 0.03}
      elbow_pitch:        {trajectory: 0.05, goal: 0.03}
      forearm_rotation:   {trajectory: 0.05, goal: 0.03}
      wrist_pitch:        {trajectory: 0.05, goal: 0.03}
      wrist_yaw:          {trajectory: 0.05, goal: 0.03}
